src/Promitor.Integrations.AzureMonitor/AzureMonitorClient.cs

```diff
@@ -217,12 +225,25 @@ private MetricValue GetMostRecentMetricValue(string metricName, TimeSeriesElemen
             return metricQuery;
         }
 
-        private IAzure CreateAzureClient(AzureEnvironment azureCloud, string tenantId, string subscriptionId, string applicationId, string applicationSecret, IOptions<AzureMonitorLoggingConfiguration> azureMonitorLoggingConfiguration, ILoggerFactory loggerFactory, MetricSinkWriter metricSinkWriter, IRuntimeMetricsCollector metricsCollector)
+        private IAzure CreateAzureClient(AzureEnvironment azureCloud, string tenantId, string subscriptionId, AuthenticationMode authenticationMode, string managedIdentityId, string applicationId, string applicationSecret, IOptions<AzureMonitorLoggingConfiguration> azureMonitorLoggingConfiguration, ILoggerFactory loggerFactory, MetricSinkWriter metricSinkWriter, IRuntimeMetricsCollector metricsCollector)
         {
-            var credentials = _azureCredentialsFactory.FromServicePrincipal(applicationId, applicationSecret, tenantId, azureCloud);
+            AzureCredentials credentials;
+
+            if (authenticationMode == AuthenticationMode.ManagedIdentity && !string.IsNullOrEmpty(managedIdentityId))
```

Comment:
Instead of making this assumption, what if `AuthenticationMode` uses `UserAssigedManagedIdentity` and `SystemAssigedManagedIdentity`?

We can then simplify the code (switch/case) and improved checks?

Reply:
The difference between UAMI and SAMI can be quite complex for a user that is not used to use them.
That's why I prefer to make the assumption regarding if user has provided the GUID.
And the "trick" is that aad podidentity "transforms" a UAMI into a virtual SAMI, and that could be disturbing at some point.
Once again, if you prefer add the 2 options, it's ok for me, but I think it will lead to more errors if the configuration is not set correctly by the user

Reply:
I'm thorn on this. I'll think about it. On one hand it's explicit with multiple options, on the other hand it requires more knowledge indeed.

Reply:
Let's split them, I'll do docs on it which should make it clear 🤷‍♂️